frame/dapps-staking/src/migrations.rs

```diff
@@ -1077,3 +1077,169 @@ pub mod v3 {
         Ok(())
     }
 }
+
+pub mod festival_end {
+
+    use super::*;
+    use frame_support::{log, storage::child::KillStorageResult, traits::Get, weights::Weight};
+    use sp_runtime::traits::Zero;
+
+    #[cfg(feature = "try-runtime")]
+    pub fn pre_migrate<T: Config>() -> Result<(), &'static str> {
+        assert!(Ledger::<T>::iter().count().is_zero());
+        assert!(BlockRewardAccumulator::<T>::exists());
+
+        Ok(())
+    }
+
+    /// Should be executed during runtime upgrade
+    /// Won't do the entire cleanup, instead it will just do necessary preparatory actions.
+    pub fn on_runtime_upgrade<T: Config>(weight_limit: Weight) -> Weight {
+        for _ in Ledger::<T>::iter() {
+            log::warn!(">>> There are existing account ledgers. This upgrade is invalid!");
+            return T::DbWeight::get().reads(1);
+        }
+
+        // 1. this needs to be handled immediately since we want to accumulate rewards in the follow-up blocks
```

Comment:
we want or don't want? (we are killing the accumulator, so I assume we don't)

Reply:
Leftover comment from previous iteration - there's actually nothing to save since we're killing the ongoing era and starting from zero.

I will modify it.